Snakefile

```diff
@@ -3,6 +3,7 @@
 # SPDX-License-Identifier: AGPL-3.0-or-later
 
 import sys
+import yaml
```

Comment:
Great draft!

I'd propose to create an auxiliary function into helpers and place there all the code.
The function may by like: check_version(config)
Internally it loads the config.default and then does the magic.
What do you think?

Reply:
Agree, revised.